app/src/main/java/com/portfolio/tasky/entry/EntryActivity.kt

```diff
@@ -33,6 +41,33 @@ class EntryActivity : AppCompatActivity(), FragmentInflater by FragmentInflaterI
     }
 
     fun setTitle(text: String) {
-        setToolBarText(viewBinding?.toolbar as Toolbar, text)
+        setToolBarText(viewBinding.toolbar, text)
+    }
+
+    override fun internetAvailable() {
+
+        runOnUiThread {
+            viewBinding.layoutOnlineMode.tvOnlineMode.let { onConnected(it, true) }
+            viewBinding.layoutOnlineMode.drawableFirst = ContextCompat.getDrawable(this, R.drawable.ic_online)
+        }
```

Comment:
Isn't this already running on ui thread?

Reply:
Yes, but it was crashing. From what I understand onAvailable and onNetworkCapabilitiesChanges from NetworkCallback sometimes run on Worker thread due to race conditions.  

Anyways with new implementation this code has been removed